edge-modules/DirectMethodSender/Program.cs

```diff
@@ -24,21 +26,47 @@ public static async Task<int> MainAsync()
             (CancellationTokenSource cts, ManualResetEventSlim completed, Option<object> handler) = ShutdownHandler.Init(TimeSpan.FromSeconds(5), Logger);
             DirectMethodSenderBase directMethodClient = null;
             ModuleClient reportClient = null;
+            Option<Uri> analyzerUrl = Settings.Current.AnalyzerUrl;
+            Option<Uri> testReportCoordinatorUrl = Settings.Current.TestResultCoordinatorUrl;
+
             try
             {
+                Guid batchId = Guid.NewGuid();
+                Logger.LogInformation($"Batch Id={batchId}");
+
                 directMethodClient = await CreateClientAsync(Settings.Current.InvocationSource);
 
-                Option<Uri> analyzerUrl = Settings.Current.AnalyzerUrl;
                 reportClient = await ModuleUtil.CreateModuleClientAsync(
                     Settings.Current.TransportType,
                     ModuleUtil.DefaultTimeoutErrorDetectionStrategy,
                     ModuleUtil.DefaultTransientRetryStrategy,
                     Logger);
-                while (!cts.Token.IsCancellationRequested)
+
+                Logger.LogInformation($"Load gen delay start for {Settings.Current.TestStartDelay}.");
+                await Task.Delay(Settings.Current.TestStartDelay);
+
+                DateTime testStartAt = DateTime.UtcNow;
+                while (!cts.Token.IsCancellationRequested && IsTestTimeUp(testStartAt))
                 {
-                    HttpStatusCode result = await directMethodClient.InvokeDirectMethodAsync(cts);
+                    (HttpStatusCode result, long dmCounter) = await directMethodClient.InvokeDirectMethodAsync(cts);
 
                     // TODO: Create an abstract class to handle the reporting client generation
+                    await testReportCoordinatorUrl.ForEachAsync(
+                        async (Uri uri) =>
+                        {
+                            TestResultCoordinatorClient trcClient = new TestResultCoordinatorClient { BaseUrl = uri.AbsoluteUri };
+                            await ModuleUtil.ReportStatus(
+                                    trcClient,
+                                    Logger,
+                                    Settings.Current.ModuleId + ".send",
+                                    ModuleUtil.FormatDirectMethodTestResultValue(
+                                        Settings.Current.TrackingId.Expect(() => new ArgumentException("TrackingId is empty")),
+                                        batchId.ToString(),
+                                        dmCounter.ToString(),
+                                        result.ToString()),
+                                    TestOperationResultType.DirectMethod.ToString());
+                        });
+
```

Comment:
For connectivity test, analyzerUrl should not have value.  Therefore below code block will call "SendEventAsync" method.    You may need to work on the todo in this PR to create an abstract class to handle reporting.

Reply:
This module is shared across E2E, LH, and Stress test. By the nature of the LH, the reporting is done using the default routing from the config file to send the report to EventHub

Reply:
For any changes added to existing code base, you need to make sure that all existing use cases work as before; and new use case should work as designed.  This part is not what it should behave in connectivity test.  You should introduce a setting parameter to distinguish what use case this module is running.  That's why @ancaantochi ask you to have an interface for reporting in this PR.

Reply:
If you think this reporting code can be shared (e.g. interface/abstract), please implement in DM projects first.  Then have another PR to refactor and reuse in other projects.